### PR TITLE
PromQL: Support trailing commas in grouping opts

### LINKED ISSUE #6470: PromQL: Consider supporting trailing commas in grouping opts
Currently PromQL allows trailing commas in label matchers, but not in the label list of grouping opts.

E.g. `a{b="c",}`,  `a{b="c"}` and `sum (a) by (b)` and are valid PromQL, while `sum (a) by (b,)` is not.

This feels somewhat inconsistent.

It might be worth considering to allow trailing commas in grouping opts, too.

### PR DESCRIPTION
**Pull request by slrtbtfs**
Fixes #6470

Signed-off-by: Tobias Guggenmos <tguggenm@redhat.com>

### CODE REVIEW

**Comment by brian-brazil** [comment]
Documentation somewhere should also change with this.

**Comment by slrtbtfs** [comment]
I've added documentation about trailing commas.

File: docs/querying/basics.md [hunk 1]
```diff
@@ -83,6 +83,10 @@ metric name that also have the `job` label set to `prometheus` and their
 
     http_requests_total{job="prometheus",group="canary"}
 
+The list of label matchers may include a trailing comma:
```
Comment (brian-brazil) [inline docs/querying/basics.md]: I don't think this justifies a whole paragraph and example. Up at line 77 we should be mentioning that the matchers are separated by commas, and then we can add a mention that trailing commas are okay.

File: docs/querying/operators.md [hunk 2]
```diff
@@ -211,6 +211,9 @@ all other labels are preserved the output. `by` does the opposite and drops
 labels that are not listed in the `by` clause, even if their label values are
 identical between all elements of the vector.
 
+`label_list` is a list of unquoted labels that may include a trailing comma, i.e.
```
Comment (brian-brazil) [inline docs/querying/operators.md]: `label list` is what we have above.

Comment (slrtbtfs) [reply docs/querying/operators.md]: I've reordered the paragraphs to a more logical order to address this.

File: docs/querying/basics.md (hunk 1 above)
Comment (slrtbtfs) [reply docs/querying/basics.md]: Thanks.

Changed this as suggested.

File: docs/querying/operators.md [hunk 3]
```diff
@@ -205,12 +205,17 @@ or
 
     <aggr-op>([parameter,] <vector expression>) [without|by (<label list>)]
 
-`parameter` is only required for `count_values`, `quantile`, `topk` and
-`bottomk`. `without` removes the listed labels from the result vector, while
+`label_list` is a list of unquoted labels that may include a trailing comma, i.e.
```
Comment (brian-brazil) [inline docs/querying/operators.md]: Above is still label list, without the underscore.

Comment (slrtbtfs) [reply docs/querying/operators.md]: Ah, now I get what you tried to tell me in the last review.

Fixed now.

File: docs/querying/operators.md [hunk 4]
```diff
@@ -205,12 +205,17 @@ or
...
-`bottomk`. `without` removes the listed labels from the result vector, while
+`label list` is a list of unquoted labels that may include a trailing comma, i.e.
+both `(label1, label2)` and `(label1, label2,)` are valid syntax.
+
+`without` removes the listed labels from the result vector, while
 all other labels are preserved the output. `by` does the opposite and drops
 labels that are not listed in the `by` clause, even if their label values are
 identical between all elements of the vector.
 
+`parameter` is only required for `count_values`, `quantile`, `topk` and
```
Comment (brian-brazil) [inline docs/querying/operators.md]: This makes it seem as though it's an option for the others. I'd say used rather than required.

Comment (slrtbtfs) [reply docs/querying/operators.md]: I personally find "required" more clear than "used" here. 

Also this wording was not introduced by this PR.

**Comment by brian-brazil** [comment]
Thanks!

**Pull request merged**